core/statedb/statedb.go

```diff
@@ -523,19 +561,60 @@ func (s *StateDB) IntermediateRoot(cleanDirty bool) error {
 		s.dirtyNftMap = make(map[int64]bool, 0)
 	}
 
+	pendingAccountItem := make([]bsmt.Item, 0, len(s.dirtyAccountsAndAssetsMap))
+	pendingNftItem := make([]bsmt.Item, 0, len(s.dirtyNftMap))
+	for i := 0; i < taskNum; i++ {
+		result := <-resultChan
+		if result.err != nil {
+			return result.err
+		}
+
+		switch result.role {
+		case accountTreeRole:
+			pendingAccountItem = append(pendingAccountItem, bsmt.Item{Key: uint64(result.index), Val: result.leaf})
+		case nftTreeRole:
+			pendingNftItem = append(pendingNftItem, bsmt.Item{Key: uint64(result.index), Val: result.leaf})
+		}
+	}
+	err := taskPool.Submit(func() {
+		resultChan <- &treeUpdateResp{
+			role: accountTreeRole,
+			err:  s.AccountTree.MultiSet(pendingAccountItem...),
+		}
+	})
+	if err != nil {
+		return err
+	}
+	err = taskPool.Submit(func() {
```

Comment:
A bit concerned about the correctness of this part, please do fullnode catch up test and prover test with 10 more tps.

Reply:
I synced with fullnode on the develop branch(no parallel) and it passed